### PR TITLE
refactor(userInfo): userinfo panel to styled component

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
^

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before
Screen Shot 2021-04-12 at 2 52 38 PM

after
Screen Shot 2021-04-12 at 2 54 26 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/profile/components/UserInfo.tsx [hunk 1]
```diff
@@ -19,17 +19,36 @@
...
-import { t } from '@superset-ui/core';
+import Collapse from 'src/common/components/Collapse';
+import { t, styled } from '@superset-ui/core';
 import { UserWithPermissionsAndRoles } from '../../types/bootstrapTypes';
 
 interface UserInfoProps {
   user: UserWithPermissionsAndRoles;
 }
 
+const StyledContainer = styled.div`
```
Comment (rusackas) [inline superset-frontend/src/profile/components/UserInfo.tsx:30]: This is failing the "shows the right user-id" unit test. It's saying "TypeError: Cannot read property 'grayscale' of undefined" which probably means the test needs to replace the standard `mount` with `import { styledMount as mount } from 'spec/helpers/theming';`

File: superset-frontend/src/profile/components/UserInfo.tsx [hunk 2]
```diff
@@ -19,17 +19,36 @@
...
+  .ant-collapse-header {
```
Comment (rusackas) [inline superset-frontend/src/profile/components/UserInfo.tsx:31]: Are all these styles unique to the implementation of this component for the UserInfo component, or are these styles likely to be reused? If they're likely to be reused, they should be added to the AntD component itself. If they're a _variant_ of the component, the AntD component being exported could use a `className` or `prop` to enable the styles. Then this UserInfo component can just use the AntD component out of the box, with some prop(s) or class(es) applied, if the use case / styles are repeatable.

Note that if you DO go that route, we should add a storybook entry to reflect the class(es)/prop(s)

File: superset-frontend/src/profile/components/UserInfo.tsx [hunk 3]
```diff
@@ -19,17 +19,36 @@
...
+    background-color: ${({ theme }) => theme.colors.grayscale.light5};
+    .header {
+      margin-left: ${({ theme }) => theme.gridUnit * -6}px;
+    }
+  }
+  .ant-collapse-content {
+    margin-top: ${({ theme }) => theme.gridUnit * -5}px;
+    border-top: none;
+  }
+  .ant-collapse
```
Comment (rusackas) [inline superset-frontend/src/profile/components/UserInfo.tsx:41]: You might not need this CSS... the AntD `collapse` component has a `showArrow` boolean prop.

Looking at the "before" version of the UI, it looks like this <Panel> component has no border, and is not collapsible. We can turn those off with `collapsible` and `bordered` props set to `false`. 

But now that I'm looking at it, I don't think we even need the Collapse component here, do we? It just looks like a white div with rounded corners is all that we need to remove the React Bootstrap component.

File: superset-frontend/src/profile/components/UserInfo.tsx [hunk 4]
```diff
@@ -42,8 +47,8 @@ export default function UserInfo({ user }: UserInfoProps) {
         />
       </a>
       <hr />
-      <Panel>
-        <Panel.Body>
+      <div className="panel">
+        <div className="header">
```
Comment (rusackas) [inline superset-frontend/src/profile/components/UserInfo.tsx:51]: Is this div even necessary? Doesn't look like it adds any styles.

Comment (pkdotson) [reply superset-frontend/src/profile/components/UserInfo.tsx:51]: Yah this is needed.

**Review by Robert8847 — APPROVED** [review]
Looks good

**Review by Robert8847 — APPROVED** [review]

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**